build.psm1

```diff
@@ -2437,21 +2437,29 @@ function Start-CrossGen {
         Generate-CrossGenAssembly -CrossgenPath $crossGenPath -AssemblyPath $assemblyPath
     }
 
-    <#
-     # The latest dotnet.exe from .NET Core 1.1.0 preview packages starts
-     # to check the existence of all TPA assemblies, including those built
-     # from local projects (e.g. powershell assemblies). The TPA assemblies
-     # in '.deps.json' are IL assemblies, so we cannot remove those ILs.
-     #
-    Write-Verbose "PowerShell Ngen assemblies have been generated, deleting ILs..." -Verbose
+    #
+    # With the latest dotnet.exe, the TPA list contains the assemblies produced by the local projects,
+    # and APP_PATHS/APP_NI_PATHS are no longer passed in. Therefore, powershell.exe starts to check the
+    # existence of all powershell assemblies, and CoreCLR by default will only load the TPAs which point
+    # to the IL assemblies. So we need to remove the ngen'ed IL assemblies and then rename the produced
+    # NI assemblies (.ni.dll) to the same as the corresponding IL assemblies.
```

Comment:
Sure we should do that. And I will update this comment to make it easy to understand.